runtime/near-vm-runner/src/logic/logic.rs

```diff
@@ -1157,23 +1157,33 @@ impl<'a> VMLogic<'a> {
                     msg: "invalid signature length".to_string(),
                 }));
             }
-            match ed25519_dalek::Signature::from_bytes(&vec) {
-                Ok(signature) => signature,
+            // Sanity-check that was performed by ed25519-dalek in from_bytes before version 2,
+            // but was removed with version 2. It is not actually any good a check, but we need
+            // it to avoid costs changing.
+            if vec[ed25519_dalek::SIGNATURE_LENGTH - 1] & 0b1110_0000 != 0 {
+                return Ok(false as u64);
+            }
+            match <&[u8; 64]>::try_from(&vec[..]) {
```

Comment:
The length check should go first (and could be rolled into the `vec.len()` comparison above.) If nothing else it:

1. statically enforces there's no panic condition in the check for `0b1110_0000`; and
2. is closer to how the code in the library was written originally (where the length check came first.)

Reply:
Also

```suggestion
            match <&[u8; ed25519_dalek::SIGNATURE_LENGTH]>::try_from(&vec[..]) {
```

?